src/Material/Menu.elm

```diff
@@ -0,0 +1,523 @@
+             Just oracle ->
+
+               let
+                 rect x y w h =
+                   String.join ""
+                   [ "rect( "
+                   , toString x ++ "px"
+                   , " "
+                   , toString y ++ "px"
+                   , " "
+                   , toString w ++ "px"
+                   , " "
+                   , toString h ++ "px"
+                   , ")"
+                   ]
+
+                 width  = oracle.menu.bounds.width
+                 height = oracle.menu.bounds.height
+               in
+                 if model.open == Just True then
+
+                     rect 0 width height 0
+
+                   else
+
+                     if model.alignment == bottomRight then
+                         rect 0 width 0 width
+                       else if model.alignment == topLeft then
+                         rect height 0 height 0
+                       else if model.alignment == topRight then
+                         rect height width height width
+                       else
+                         ""
+          )
+          (isJust model.oracle)
+        ]
+        ( let
+
+            makeItem n item =
+              let
+                transitionDuration =
+                  constant.transitionDurationSeconds *
+                  constant.transitionDurationFraction
+
+                offsetTop n =
+                  case model.oracle of
+                    Nothing -> 0
+                    Just oracle ->
+                      oracle.offsetTops
+                      |> List.drop (n-1) -- n is 1-based
+                      |> List.head
+                      |> fromJust "Menu.view: offsetTop"
+
+                offsetHeight n =
+                  case model.oracle of
+                    Nothing -> 0
+                    Just oracle ->
+                      oracle.offsetHeights
+                      |> List.drop (n-1) -- n is 1-based
+                      |> List.head
+                      |> fromJust "Menu.view offsetHeight"
+
+                height =
+                  case model.oracle of
+                    Nothing -> 0
+                    Just oracle -> oracle.menu.bounds.height
+
+                itemDelay =
+
+                  if model.alignment == topLeft ||
+                     model.alignment == topRight then
+
+                      (height - offsetTop n - offsetHeight n) / height * transitionDuration
+                      |> toString
+                      |> flip (++) "s"
+
+                    else
+
+                      ((offsetTop n) / height * transitionDuration)
+                      |> toString
+                      |> (flip (++) "s")
+              in
+
+                    styled li
+                    ( (if item.enabled then
+                          Style.attribute (onClick addr Oracle.decode' (Close' n))
+                        else
+                          Style.attribute (Html.attribute "disabled" "disabled"))
```

Comment:
Is there some reasons you don't use `Html.Attribute.disabled`? (I ask because I had trouble with it in the past and hope you have shown me a solution.)